### PR TITLE
Added profile page

### PR DESCRIPTION
**Pull request by disjfa**
I seup a default profile page.

Also i added a article list component for use for all components.

### CODE REVIEW

File: src/views/Profile.vue [hunk 1]
```diff
@@ -1,20 +1,21 @@
...
-            <button class="btn btn-sm btn-outline-secondary action-btn">
+            <button class="btn btn-sm btn-secondary action-btn" v-if="profile.following" @click.prevent="unfollow()">
+              <i class="ion-checkmark-round"></i>
+              Following {{ profile.username }}
+            </button>
+            <button class="btn btn-sm btn-outline-secondary action-btn" v-if="!profile.following" @click.prevent="follow()">
               <i class="ion-plus-round"></i>
-              &nbsp;
-              Follow Eric Simons
+              Follow {{ profile.username }}
```
Comment (vilsbole) [inline src/views/Profile.vue:18]: There is just missing the third option:
When the connected user is looking at his own page, the follow button becomes a link to his setting profile

screen shot 2017-11-10 at 21 53 54

File: src/components/ArticleList.vue [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        required: false
+      }
+    },
+    data () {
+      return {
+        currentPage: 1
+      }
+    },
+    computed: {
+      listConfig () {
```
Comment (vilsbole) [inline src/components/ArticleList.vue:58]: Would a switch case be useful here?

**Review by vilsbole — CHANGES_REQUESTED** [review]
Great job! There is just missing a minor use case.

File: src/components/ArticleList.vue (hunk 2 above)
Comment (disjfa) [reply src/components/ArticleList.vue:58]: Nope, they are 3 diferent api parameters. So to keep the 3 parameters is useful

File: src/views/Profile.vue (hunk 1 above)
Comment (disjfa) [reply src/views/Profile.vue:18]: Yes, did not want to change the auth store to help with that but it has to be done.

Comment (vilsbole) [reply src/views/Profile.vue:18]: Ok, I'll merge it now and wait for your next PR

**Pull request merged**